controllers/vspherecluster_controller.go

```diff
@@ -230,6 +217,7 @@ func (r clusterReconciler) reconcileDelete(ctx *context.ClusterContext) (reconci
 		return reconcile.Result{}, err
 	}
 	if len(haproxyLoadbalancers.Items) > 0 {
+		conditions.MarkFalse(ctx.VSphereCluster, infrav1.LoadBalancerAvailableCondition, clusterv1.DeletingReason, clusterv1.ConditionSeverityInfo, "")
```

Comment:
Should we add the following after line 225? To flip the condition reason from Deleting to Deleted once the loadbalancers have been successfully removed

```
if conditions.Has(ctx.VSphereCluster, infrav1.LoadBalancerAvailableCondition) {
	conditions.MarkFalse(ctx.VSphereCluster, infrav1.LoadBalancerAvailableCondition, clusterv1.DeletedReason, clusterv1.ConditionSeverityInfo, "")
}
```

Reply:
Done
However, in a normal delete workflow immediately after the Load balancer is deleted the finalizer gets removed, so the change won't be visible to the users.

Reply:
I was thinking of a case where the vsphereMachines aren't getting deleted properly but the loadbalancer was successfully deleted. Unsure how probable / common that failure is though.

Reply:
I see a potential bug in the logic here, because we should not start deleting the load balancer if there are still machines, otherwise the cluster is busted (and if I'm not wrong cluster controller in CAPI has this control)

Reply:
I agree with @fabriziopandini - this might turn into a potential bug when we start deleting in-cluster objects (e.g. `Services`, `PVCs`...), but for now we don't require `kube-apiserver` access to complete the deletion.

@fabriziopandini - can you file an issue for this ?

Reply:
I agree with @fabriziopandini - this might turn into a potential bug when we start deleting in-cluster objects (e.g. `Services`, `PVCs`...), but for now we don't require `kube-apiserver` access to complete the deletion.

@fabriziopandini - can you file an issue for this ?

Reply:
done, see https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/issues/1033